src/hypertable.c

```diff
@@ -812,6 +812,16 @@ table_has_tuples(Oid table_relid, Snapshot snapshot, LOCKMODE lockmode)
 	return hastuples;
 }
 
+static bool
+table_is_logged(Oid table_relid)
+{
+	Relation	rel = RelationIdGetRelation(table_relid);
+	bool		res = (rel->rd_rel->relpersistence == RELPERSISTENCE_PERMANENT);
```

Comment:
Btw, there's this function called `get_rel_persistence()` that uses the syscache. Maybe this is what should be used here?